lib/erb_lint/linters/content_style.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+
+module ERBLint
+  class Linter
+    # Checks for content style guide violations in the text nodes of HTML files.
+    class ContentStyle < Linter
+      include LinterRegistry
+
+      def initialize(config)
+        @content_ruleset = []
+        config.fetch('rule_set', []).each do |rule|
+          suggestion = rule.fetch('suggestion', '')
+          regex_description = rule.fetch('regex_description', '')
+          case_insensitive = rule.fetch('case_insensitive', false)
+          violation = rule.fetch('violation', [])
+          (violation.is_a?(String) ? [violation] : violation).each do |violating_pattern|
+            @content_ruleset.push(
+              violating_pattern: violating_pattern,
+              suggestion: suggestion,
+              case_insensitive: case_insensitive,
+              regex_description: regex_description
+            )
+          end
+        end
+        @content_ruleset.freeze
+        @addendum = config.fetch('addendum', '')
+      end
+
+      def lint_file(file_tree)
+        errors = []
+        @prior_violations = []
+        all_text = Parser.get_text_nodes(file_tree)
+        all_text.each do |text_node|
+          # Next if node doesn't contain content
+          next if text_node.text =~ /\A(\n|\s)*\z/
```

Comment:
```ruby
next unless text_node.text =~ /[^\n\s]/
```

inverting the statement makes it simpler: "move to the next unless text has a content character".